cli/src/main/java/com/box/l10n/mojito/cli/command/checks/ContextAndCommentCliChecker.java

```diff
@@ -108,47 +125,59 @@ assetExtractorTextUnit, checkTextUnit(assetExtractorTextUnit)))
   }
 
   private ContextAndCommentCliCheckerResult getContextAndCommentCliCheckerResult(
-      AssetExtractorTextUnit assetExtractorTextUnit, String failureText) {
+      AssetExtractorTextUnit assetExtractorTextUnit, CliCheckResult.CheckFailure checkFailure) {
     ContextAndCommentCliCheckerResult result;
-    if (failureText != null) {
+    if (checkFailure != null) {
       logger.debug(
           "'{}' source string failed check with error: {}",
           assetExtractorTextUnit.getSource(),
-          failureText);
+          checkFailure.failureMessage());
       result =
           new ContextAndCommentCliCheckerResult(
-              true, assetExtractorTextUnit.getSource(), failureText);
+              true,
+              assetExtractorTextUnit.getSource(),
+              checkFailure.failureMessage(),
+              checkFailure.ruleId());
     } else {
       result = new ContextAndCommentCliCheckerResult(false);
     }
     return result;
   }
 
-  private String checkTextUnit(AssetExtractorTextUnit assetExtractorTextUnit) {
+  private CliCheckResult.CheckFailure checkTextUnit(AssetExtractorTextUnit assetExtractorTextUnit) {
     String failureText = null;
     String[] splitNameArray = assetExtractorTextUnit.getName().split("---");
     String context = null;
     if (isPlural(assetExtractorTextUnit) && cliCheckerOptions.getPluralsSkipped()) {
-      return failureText;
+      return null;
     }
     if (splitNameArray.length > 1) {
       context = splitNameArray[1];
     }
     String comment = assetExtractorTextUnit.getComments();
+    String ruleId = null;
 
     if (!isBlank(context) && !isBlank(comment)) {
       if (context.trim().equalsIgnoreCase(comment.trim())) {
+        ruleId = "EQUAL_CONTEXT_AND_COMMENT_STRINGS";
         failureText = "Context & comment strings should not be identical.";
       }
     } else if (isBlank(context) && isBlank(comment)) {
+      ruleId = "EMPTY_CONTEXT_AND_COMMENT_STRINGS";
       failureText = "Context and comment strings are both empty.";
     } else if (isBlank(context)) {
+      ruleId = "EMPTY_CONTEXT_STRING";
       failureText = "Context string is empty.";
     } else if (isBlank(comment)) {
+      ruleId = "EMPTY_COMMENT_STRING";
```

Comment:
Ah I see why you can't re-use the CliCheckerType enum as some checks have multiple rule ids, moving these to their own enum will centralise our rule ids and make them easier to add/track in future